[docs] Updating steps to connect to the dev server for apple devices (#23616)
expo/expo: docs/pages/get-started/expo-go.mdx

## How it works

> Expo Go is a native app that is installed on your device. When you run `npx expo start` in your project, [Expo CLI](/more/expo-cli/) starts a [development server](/more/expo-cli/#develop) and generates a QR code. You can then open the Expo Go app on your device and scan the QR code to connect to the dev server. If you are running the app on iOS, use the device's camera app to scan the QR code.

Reviewer: ```suggestion
Expo Go is a native app that is installed on your device. When you run `npx expo start` in your project, [Expo CLI](/more/expo-cli/) starts a [development server](/more/expo-cli/#develop) and generates a QR code. On Android, you can open the Expo Go app on your device and scan the QR code to connect to the dev server. On iOS, use the device's camera to scan the QR code.
```
Adding the context about Android to make the difference clear as to when to use the Expo Go app and when to use device camera.
